botocore/paginate.py

```diff
@@ -191,6 +191,8 @@ def _handle_first_request(self, parsed, primary_result_key,
         # the truncated amount.
         starting_truncation = self._parse_starting_token()[1]
         all_data = primary_result_key.search(parsed)
+        if all_data is None:
+            all_data = []
```

Comment:
Where did we land on the default value for this?

In this scenario the result key actually evaluates to a string, so defaulting to an empty list does not seem like the correct behavior.

What's our plan to address this?